src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs

```diff
@@ -149,61 +150,96 @@ public bool AddToCache (AssemblyDefinition assembly)
 
 		protected virtual AssemblyDefinition ReadAssembly (string file)
 		{
-			bool haveDebugSymbols = loadDebugSymbols &&
-				(File.Exists (Path.ChangeExtension (file, ".pdb")) ||
-				 File.Exists (file + ".mdb"));
 			var reader_parameters = new ReaderParameters () {
 				ApplyWindowsRuntimeProjections  = loadReaderParameters.ApplyWindowsRuntimeProjections,
 				AssemblyResolver                = this,
 				MetadataImporterProvider        = loadReaderParameters.MetadataImporterProvider,
 				InMemory                        = loadReaderParameters.InMemory,
 				MetadataResolver                = loadReaderParameters.MetadataResolver,
 				ReadingMode                     = loadReaderParameters.ReadingMode,
-				ReadSymbols                     = haveDebugSymbols,
 				ReadWrite                       = loadReaderParameters.ReadWrite,
 				ReflectionImporterProvider      = loadReaderParameters.ReflectionImporterProvider,
 				SymbolReaderProvider            = loadReaderParameters.SymbolReaderProvider,
-				SymbolStream                    = loadReaderParameters.SymbolStream,
 			};
 			try {
 				return LoadFromMemoryMappedFile (file, reader_parameters);
 			} catch (Exception ex) {
 				logger (
 						TraceLevel.Verbose,
 						$"Failed to read '{file}' with debugging symbols. Retrying to load it without it. Error details are logged below.");
-				logger (TraceLevel.Verbose, $"{ex.ToString ()}");
+				logger (TraceLevel.Verbose, ex.ToString ());
 				reader_parameters.ReadSymbols = false;
 				return LoadFromMemoryMappedFile (file, reader_parameters);
+			} finally {
+				reader_parameters.SymbolStream?.Dispose ();
 			}
 		}
 
 		AssemblyDefinition LoadFromMemoryMappedFile (string file, ReaderParameters options)
 		{
 			// We can't use MemoryMappedFile when ReadWrite is true
 			if (options.ReadWrite) {
+				if (loadDebugSymbols) {
+					LoadSymbols (file, options, File.OpenRead);
+				}
 				return AssemblyDefinition.ReadAssembly (file, options);
 			}
 
-			MemoryMappedViewStream? viewStream = null;
+			// Likely 6 entries when symbols exist
```

Comment:
Where does "6" come from? Why is that likely?

Reply:
Looks like `OpenMemoryMappedViewStream()` has 3 `disposables.Add()` calls, so a multiple of 3 makes sense, and there are potentially 2 `OpenMemoryMappedViewStream()` invocations here.

That said, the feels "distant"; if (when) `OpenMemoryMappedViewStream()` changes, this value will be "wrong".  (Not necessarily a bad thing, but…)

I *think* I'd prefer:

```csharp
AssemblyDefinition LoadFromMemoryMappedFile (…)
{
    // …
    var disposables = new List<IDisposable>(
        (1 + (loadDebugSymbols ? 1 : 0)) * OpenMemoryMappedViewStream_disposables_Add_calls);
    // …
}

// Number of times OpenMemoryMappedViewStream() calls disposables.Add()
const int OpenMemoryMappedViewStream_disposables_Add_calls = 3;
static MemoryMappedViewStream OpenMemoryMappedViewStream (string file, List<IDisposable> disposables)
{
    // …
}
```

Reply:
I made these changes.